lib/utils/getFormatter.mjs

```diff
@@ -0,0 +1,64 @@
+import { EXIT_CODE_INVALID_CONFIG } from '../constants.mjs';
+import dynamicImport from './dynamicImport.mjs';
+import { existsSync } from 'node:fs';
+import formatters from '../formatters/index.mjs';
+import getConfigForFile from '../getConfigForFile.mjs';
+import getFormatterOptionsText from './getFormatterOptionsText.mjs';
+import { resolve } from 'node:path';
+
+/** @import {Formatter, FormatterType, InternalApi} from 'stylelint' */
+
+/**
+ * @param {InternalApi} stylelint
+ * @param {Object} [params]
+ * @param {FormatterType | Formatter | undefined} [params.formatterType]
+ * @returns {Promise<Formatter>}
+ */
+export default async function getFormatter(stylelint) {
+	const cwd = stylelint._options.cwd;
+	const configPath = stylelint._options.configFile || cwd;
+	let formatter = stylelint._options.formatter;
+
+	if (!formatter) {
+		let configForFile;
+
+		try {
+			configForFile = await getConfigForFile(stylelint, configPath);
+		} catch (err) {
+			if (err instanceof Error && 'code' in err && err.code === EXIT_CODE_INVALID_CONFIG) {
+				configForFile = undefined;
```

Comment:
[suggestion] This line seems to be uncovered by test cases. Can you add a test case to `standalone-formatter.test.mjs` when a config file doesn't exist and the `formatter` argument is unspecified?